### PR TITLE
Add custom error log file

### PR DESCRIPTION
**Pull request by alanj853**
Saves having to parse kmsg on start up

### CODE REVIEW

File: src/heart.c [hunk 1]
```diff
@@ -383,12 +385,21 @@ kill_old_erlang(void)
     }
 }
 
+static void write_to_custom_error_log(int reason) {
+    FILE *fp;
+    fp = fopen(custom_error_log, "w+");
```
Comment (metelik) [inline src/heart.c]: Please. Please.. not to the /var/system/.. We have a directory here: /var/system/priv/log/

File: src/heart.c [hunk 2]
```diff
@@ -158,6 +159,7 @@ static int  wait_until_close_write_or_env_tmo(int);
 /*  static variables */
 
 static const char *watchdog_path = "/dev/watchdog0";
+static const char *custom_error_log = "/var/system/nerves_heart_error.log";
```
Comment (metelik) [inline src/heart.c]: instead of hardcoding it we could provide a command line argiment like -log see get_arguments function

**Review by metelik — COMMENTED** [review]
Looks good. We could improve it hither and tither -see my comments an - and perhaps submit improvements to the upstream.

**Review by metelik — APPROVED** [review]

File: src/heart.c (hunk 2 above)
Comment (alanj853) [reply src/heart.c]: This is what I wanted to do, but that is not as trivial as it looks. If you look at that get_arguments functions, it's not clear to me how the arguments he parses for match up to the ones you have to give it in the README. He parse for '-h' and '-p', yet in the README, he mentions only how to add a HEART_BEAT_TIMEOUT.

File: src/heart.c (hunk 1 above)
Comment (alanj853) [reply src/heart.c]: Okay fair enough. Forgot about that.

File: src/heart.c (hunk 2 above)
Comment (metelik) [reply src/heart.c]: yup wonders of ye open-source :D documentatio is ebing put on a long finger

**Review by metelik — APPROVED** [review]
👍

**Pull request merged**